archive/terraform/main.tf

```diff
@@ -55,3 +55,53 @@ module "registrar" {
   source_queue_name = "${module.registrar_queue.name}"
   source_queue_arn  = "${module.registrar_queue.arn}"
 }
+
+module "bagger" {
+  source = "service"
+
+  service_egress_security_group_id = "${aws_security_group.service_egress_security_group.id}"
+  cluster_name                     = "${aws_ecs_cluster.cluster.name}"
+  namespace_id                     = "${aws_service_discovery_private_dns_namespace.namespace.id}"
+  subnets                          = "${local.private_subnets}"
+  vpc_id                           = "${local.vpc_id}"
+  service_name                     = "${local.namespace}_bagger"
+  aws_region                       = "${var.aws_region}"
+
+  min_capacity = 1
+  max_capacity = 1
+
+  env_vars = {
+    METS_BUCKET_NAME            = "${var.bagger_mets_bucket_name}"
+    READ_METS_FROM_FILESHARE    = "${var.bagger_read_mets_from_fileshare}"
+    WORKING_DIRECTORY           = "${var.bagger_working_directory}"
+    DROP_BUCKET_NAME            = "${var.bagger_drop_bucket_name}"
+    DROP_BUCKET_NAME_METS_ONLY  = "${var.bagger_drop_bucket_name_mets_only}"
+    DROP_BUCKET_NAME_ERRORS     = "${var.bagger_drop_bucket_name_errors}"
+    CURRENT_PRESERVATION_BUCKET = "${var.bagger_current_preservation_bucket}"
+    DLCS_SOURCE_BUCKET          = "${var.bagger_dlcs_source_bucket}"
+    BAGGING_QUEUE               = "${module.bagger_queue.name}"
+
+    # aws creds that can see Preservica as well as Platform estate
+    AWS_ACCESS_KEY_ID     = "${var.bagger_aws_access_key_id}"
+    AWS_SECRET_ACCESS_KEY = "${var.bagger_aws_secret_access_key}"
+    AWS_DEFAULT_REGION    = "${var.aws_region}"
+
+    # DLCS config
+    DLCS_ENTRY       = "${var.bagger_dlcs_entry}"
+    DLCS_API_KEY     = "${var.bagger_dlcs_api_key}"
+    DLCS_API_SECRET  = "${var.bagger_dlcs_api_secret}"
+    DLCS_CUSTOMER_ID = "${var.bagger_dlcs_customer_id}"
+    DLCS_SPACE       = "${var.bagger_dlcs_space}"
+
+    # DDS credentials
+    DDS_API_KEY      = "${var.bagger_dds_api_key}"
+    DDS_API_SECRET   = "${var.bagger_dds_api_secret}"
+    DDS_ASSET_PREFIX = "${var.bagger_dds_asset_prefix}"
+  }
```

Comment:
Similar here, we should check with @kenoir whether it would be better to manage these values in another way.

Reply:
@mklander These keys are for an external service (the DLCS), so it won't be possible to carry them by IAM role.